lib/rules/curly.js

```diff
@@ -290,7 +290,9 @@ module.exports = {
                 }
             } else if (multiOrNest) {
                 if (hasBlock && body.body.length === 1 && isOneLiner(body.body[0])) {
-                    expected = false;
+                    const leadingComments = sourceCode.getComments(body.body[0]).leading;
+
+                    expected = !!leadingComments.length;
```

Comment:
Can you use `leadingComments.length > 0` instead? I prefer to be explicit about the expectation here.